physionet-django/console/views.py

```diff
@@ -2891,3 +2892,96 @@ def event_management(request, event_slug):
         {
             'event': selected_event
         })
+
+
+@permission_required('events.add_event', raise_exception=True)
+def event_agreement_list(request):
+    if request.method == 'POST':
+        event_agreement_form = EventAgreementForm(data=request.POST)
+        if event_agreement_form.is_valid():
+            event_agreement_form.instance.creator = request.user
+            event_agreement_form.save()
+            event_agreement_form = EventAgreementForm()
+            messages.success(request, "The Event Agreement has been created.")
+        else:
+            messages.error(request, "Invalid submission. Check errors below.")
+    else:
+        event_agreement_form = EventAgreementForm()
+
+    event_agreements = EventAgreement.objects.filter(creator=request.user).order_by('name')
+    event_agreements = paginate(request, event_agreements, 20)
+
+    return render(
+        request,
+        'console/event_agreement_list.html',
+        {
+            'event_agreement_nav': True,
+            'event_agreements': event_agreements,
+            'event_agreement_form': event_agreement_form
+        }
+    )
+
+
+@permission_required('events.add_event', raise_exception=True)
+def event_agreement_detail(request, pk):
+    event_agreement = get_object_or_404(EventAgreement, pk=pk)
+
+    if request.method == 'POST':
+        event_agreement_form = EventAgreementForm(data=request.POST, instance=event_agreement)
+        if event_agreement_form.is_valid():
+            event_agreement_form.save()
+            messages.success(request, "The Event Agreement has been updated.")
+        else:
+            messages.error(request, "Invalid submission. Check errors below.")
+
+    else:
+        event_agreement_form = EventAgreementForm(instance=event_agreement)
+
+    return render(
+        request,
+        'console/event_agreement_detail.html',
+        {
+            'event_agreement_nav': True,
+            'event_agreement': event_agreement,
+            'event_agreement_form': event_agreement_form
+        }
+    )
+
+
+@permission_required('events.add_event', raise_exception=True)
+def event_agreement_delete(request, pk):
+    if request.method == 'POST':
+        event_agreement = get_object_or_404(EventAgreement, pk=pk)
+        event_agreement.delete()
```

Comment:
add a message telling the user the event is deleted.
```suggestion
        event_agreement.delete()
        messages.success(request, "The Event Agreement has been deleted.")
```